Tag System.ProcessID from EventData for network events.

scudette opened the pull request:
Network Events report the responsible pid in the EventData - for consistancy we want the pid to be accessible in the System properties.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Velocidex/etw?pullRequest=2) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Velocidex/etw?pullRequest=2) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Velocidex/etw?pullRequest=2) it.</sub>

zaneGittins commented:
Signed the CLA but seems to be stuck pending.

scudette commented:
This should already work and we can pass in the keywords from the watch_etw() plugin (using any and all parameters).

zaneGittins commented:
Got it, thank you! Would you mind giving an example of watch_etw using any/all VQL ? I see that they are type uint64, but not sure what to pass. Hoping to monitor Kernel FileIoInit.

scudette commented:
Certainly, here is an example where we report file activity and tie it with the process chain

https://github.com/Velocidex/velociraptor/blob/master/artifacts/definitions/Windows/ETW/FileCreation.yaml#L32

That provider gives a lot of events types and so it's necessarily to limit them by keywords

The any parameter is the logical or of all the keywords you want to include in the stream

zaneGittins commented:
I could be mistaken, but I think the kernel logger is a little different, it's a special GUID `9e814aad-3204-11d2-9a82-006008a86939`  (NT Kernel Logger) and the Windows API calls needed to read events from it are slightly different. For example, EnableTraceEx2 isn't called when subscribing to `9e814aad-3204-11d2-9a82-006008a86939`.

https://learn.microsoft.com/en-us/windows/win32/etw/configuring-and-starting-the-nt-kernel-logger-session

bmcder02 commented:
Hey @zaneGittins , The Kernel Logger Session is a predefined list of kernel providers. This can be accomplished with more flexibility, by directly calling the Kernel providers yourself. There's a few extra Kernel providers on newer versions of Windows, so all the more excitement. 
I've made a change to this library (#3 ) so that we can now request multiple providers within a session, with a slight change in the order of operations (session update is now required). A change will also be made to Velociraptor so that multiple providers can be called at the same time, so this can become an artifact.

zaneGittins commented:
@bmcder02 That sounds awesome! Thank you! I'll close this pull request.

scudette committed:
Tag System.ProcessID from EventData for network events.

Network Events report the responsible pid in the EventData - for
consistancy we want the pid to be accessible in the System properties.
